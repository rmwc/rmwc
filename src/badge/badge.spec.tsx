import React from 'react';
import { render, screen } from '@testing-library/react';
import { Badge, BadgeAnchor } from '.';

describe('Badge', () => {
  it('renders with label', () => {
    const { asFragment } = render(<Badge label="New" />);

    expect(screen.getByText('New')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('align end', () => {
    const { container } = render(<Badge align="end" />);

    expect(container.firstChild).toHaveClass('rmwc-badge--align-end');
  });

  it('align start', () => {
    const { container } = render(<Badge align="start" />);

    expect(container.firstChild).toHaveClass('rmwc-badge--align-start');
  });

  it('exited', () => {
    const { container } = render(<Badge exited />);

    expect(container.firstChild).toHaveClass('rmwc-badge--exited');
  });

  it('inset', () => {
    // cant assert, jsdom doesn't include css vars?
    const { asFragment } = render(<Badge inset="10rem" />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('BadgeAnchor', () => {
  it('renders', () => {
    const { asFragment } = render(<BadgeAnchor />);

    expect(asFragment()).toMatchSnapshot();
  });
});
