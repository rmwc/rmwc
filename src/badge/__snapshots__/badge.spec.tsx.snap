// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge inset 1`] = `
<DocumentFragment>
  <div
    class="rmwc-badge rmwc-badge--align-end rmwc-badge--no-content"
    style="--rmwc-badge-inset: 10rem;"
  >
    Â 
  </div>
</DocumentFragment>
`;

exports[`Badge renders with label 1`] = `
<DocumentFragment>
  <div
    class="rmwc-badge rmwc-badge--align-end"
  >
    New
  </div>
</DocumentFragment>
`;

exports[`BadgeAnchor renders 1`] = `
<DocumentFragment>
  <div
    class="rmwc-badge-anchor"
  />
</DocumentFragment>
`;
